pkg/controller/shared_state.go

```diff
@@ -1564,3 +1571,60 @@ func (c *SharedState) getNodesTopologySelector(ctx context.Context, nodes []v1.N
 	}
 	return topologySelector
 }
+
+func (s *SharedState) addNodeToAvailableAttachmentsMap(ctx context.Context, node string) {
+	nodeObj := &v1.Node{}
+	if err := s.cachedClient.Get(ctx, types.NamespacedName{Name: node}, nodeObj); err != nil {
+		capacity, _ := azureutils.GetNodeMaxDiskCountWithLabels(nodeObj.Labels)
+		var count atomic.Int32
+		count.Store(int32(capacity))
+		s.availableAttachmentsMap.Store(node, &count)
+		return
+	}
+	capacity, _ := azureutils.GetNodeRemainingDiskCount(ctx, s.cachedClient, node)
+	var count atomic.Int32
+	count.Store(int32(capacity))
+	s.availableAttachmentsMap.Store(node, &count)
```

Comment:
You should check to see if the entry is already in the map before unconditionally adding it. Although this change may ensure that this isn't the case through the reconciler watch predicate, future changes may break this assumption.

```go
if _, ok := s.availableAttachmentsMap.Load(node); !ok {
    capacity, _ := azureutils.GetNodeRemainingDiskCount(ctx, s.cachedClient, node)
    var count atomic.Int32
    count.Store(int32(capacity))
    _, _ := s.availableAttachmentsMap.LoadOrStore(node, &count)
```

Adding the count using `LoadOrStore` instead of just `Store` handle races where this function is called concurrently for the same node.

Reply:
Done.